### ISSUE TITLE
distill: train the offline SFT baseline that every on-policy variant must beat

### ISSUE THREAD

**Issue by asuzukosi** [asuzukosi/ai-experiments#29]
Supervised fine-tuning on the 50K teacher completions. **This is the line every on-policy variant has to beat**, and it is a baseline, not a scaling study — one data size, not three.

**Epoch count is a decision, not a default, and it interacts with teacher strength.** A strong teacher peaks at epoch 1 (45.0%) then declines (43.8%, 44.2%), while a weaker teacher keeps improving across three (30.6% -> 32.4% -> 33.6%). With a 32B teacher one epoch is right, but evaluate a 2-epoch checkpoint on at least one arm to confirm the peak is where we assume — that costs an eval pass, not a retrain.

**Do not select checkpoints on validation loss.** Omitting `<think>` tokens produced *lower* validation loss and *worse* MATH-500 accuracy in almost every published row. Loss and task metric decouple here.

### Sketch

```python
# src/distill/kd.py
from trl import SFTConfig, SFTTrainer

cfg = SFTConfig(
    output_dir=f"results/checkpoints/{run_name('qwen2.5-1.5b', 'offline-sft', now)}",
    num_train_epochs=1,                 # 32B teacher -> 1 epoch; verify with a 2-epoch eval
    learning_rate=1e-5,
    packing=True,
    completion_only_loss=True,          # prompt masking
    max_length=1024,
    save_strategy="steps", save_steps=50, save_total_limit=3,
    report_to="wandb",
)
trainer = SFTTrainer(model="Qwen/Qwen2.5-1.5B", args=cfg, train_dataset=teacher_data)
trainer.train(resume_from_checkpoint=True)

# checkpoint selection is on the TASK metric, never on loss
best = max(checkpoints, key=lambda c: gsm8k_pass1(c))
```

### TODO
- [ ] SFT on 50K teacher completions: prompt masking, packing, lr 1e-5, 1 epoch
- [ ] Evaluate a 2-epoch checkpoint on at least one arm to confirm the peak
- [ ] **Select checkpoints on the task metric, never on validation loss**
- [ ] Fix `<think>` handling once and apply it identically across all four variants — a silent 1-4 pt confound
- [ ] Record tokens trained and surviving example count (needed for the equal-token comparison)

### References
- TRL SFTTrainer: https://huggingface.co/docs/trl/main/en/sft_trainer
- Sequence-level knowledge distillation (Kim & Rush): https://arxiv.org/abs/1606.07947
- Qwen2.5-1.5B: https://huggingface.co/Qwen/Qwen2.5-1.5B

### Compute

1x RTX 4090 24GB. ~2 hours.

---

### Dependencies

- **Blocked by:** #28
- **Blocks:** #33

Self-contained within this project — no cross-project dependencies.
